### ISSUE TITLE
[java] ArrayIsStoredDirectly reports duplicated violation

### ISSUE THREAD

**Issue by MetaBF** [pmd/pmd#3879]
**Affects PMD Version: 6.44.0** 

**Rule: ArrayIsStoredDirectly**
https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#arrayisstoreddirectly 

**Description:** 
PMD alert twice at line 5 in the following example. I think a single warning at line 7 is better since it's where the store operation takes place.  

**Code Sample demonstrating the issue:**

```java
public class ExampleAISD { 

    public String[] strs;

    public void func(String[] strs, boolean flag) { // line 5
        if (flag) {
            this.strs = strs; // line 7
        }
    }
}
```

**Expected outcome:**

PMD reports twice at line 5, but that's wrong.  

**Running PMD through:** *[CLI]*

**Comment by MetaBF** [comment]
@oowekyala

**Comment by adangel** [comment]
I think, we have there two problems:
1) I agree, it would be more useful, if the rule would report the violation on line 7. The current code always reports the formal parameter. That would be an improvement
2) We seem to be confused by the field name - which is the same as the parameter name: `strs`...
